### PR TITLE
[New practice exercise]: State of Tic-Tac-Toe

### PR DESCRIPTION
**Pull request by ryanplusplus**

### CODE REVIEW

File: exercises/practice/state-of-tic-tac-toe/src/state_of_tic_tac_toe.cr [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  enum Error
+    XWentTwice
+    OStarted
+    GameShouldHaveEndedAfterTheGameWasWon
+  end
+
+  private alias Board = Array(String)
+
+  # Write your code for the 'Run Length Encoding' exercise in this file.
```
Comment (Steffan153) [inline exercises/practice/state-of-tic-tac-toe/src/state_of_tic_tac_toe.cr:16]:
```suggestion
  # Write your code for the 'State of Tic-Tac-Toe' exercise in this file.
```

Comment (ryanplusplus) [reply exercises/practice/state-of-tic-tac-toe/src/state_of_tic_tac_toe.cr]: Thanks!

**Review by meatball133 — COMMENTED** [review]
I think this looks really good, I really like that we can get an exercise that utilizes enum in a good way.

Although I think perhaps a `.docs/instructions.append.md` could be handy to explain that.

Here is a draft of what I think it could look like, but feel free to write it as you want:

> In this exercise you are given 2 enums.
  You are meant to utilize the enums as the return values for the method `state`.
  You can read more about enums here: [enum][enum]
[enum]: https://crystal-lang.org/reference/latest/syntax_and_semantics/enum.html

**Comment by ryanplusplus** [comment]
> I think this looks really good, I really like that we can get an exercise that utilizes enum in a good way.
> 
> Although I think perhaps a `.docs/instructions.append.md` could be handy to explain that.
> 
> Here is a draft of what I think it could look like, but feel free to write it as you want:
> 
> > In this exercise you are given 2 enums.
> > You are meant to utilize the enums as the return values for the method `state`.
> > You can read more about enums here: [enum][enum]
> > [enum]: https://crystal-lang.org/reference/latest/syntax_and_semantics/enum.html

Thanks, this was a good idea. I took your suggestion and amended it to also cover union types.

**Comment by meatball133** [comment]
Hi, I did not see another commit, did you forget to push or are you still working on it?

**Comment by ryanplusplus** [comment]
> Hi, I did not see another commit, did you forget to push or are you still working on it?

Yep, sorry about that. There was a non fast forward that I didn't notice. Thanks for the reminder.

**Review by meatball133 — APPROVED** [review]
Awesome!

April is ending, and your work on doing these exercises is fantastic!

I still have some small things left but otherwise, this month had some progress I have been happy with like the move to Crystal 1.8.

For next month (May) so will I personally be quite busy, it is the busiest month of the year for me. But that doesn't mean I will not be reviewing prs. It mostly means I won't be creating as much content but it could also mean there is some shorter period I won't be reviewing ( shouldn't be longer than 1-2 days periods).

In any case for next month if you still like to contribute, so have I talked with Jeremy and if you would like you could become a co-maintainer of the track since you have regularly contributed to the track. It doesn't really hold on to any obligations but in general, it is about maintaining and deciding how the track should move on. Although I will keep it to myself to l be the person writing the monthly blog posts (which I post on the forum). 

It is totally up to you if you like to be a maintainer or if you like to just keep being a contributor or if you would like to take a pause in contributing.

**Comment by ryanplusplus** [comment]
> I still have some small things left but otherwise, this month had some progress I have been happy with like the move to Crystal 1.8 and 5 new exercises.

Did we get 5 new exercises? I think I've done 3 (Say, Diamond, Tic-Tac-Toe) and you've done 1 (Bank Account) and I was planning to do the fifth (probably Matrix) this weekend. If you want to push Matrix to next month that's fine with me.

> For next month (May) so will I personally be quite busy, it is the busiest month of the year for me. But that doesn't mean I will not be reviewing prs. It mostly means I won't be creating as much content but it could also mean there is some shorter period I won't be reviewing ( shouldn't be longer than 1-2 days periods).

Sure, that's no big deal.

> In any case for next month if you still like to contribute, so have I talked with Jeremy and if you would like you could become a co-maintainer of the track since you have regularly contributed to the track. It doesn't really hold on to any obligations but in general, it is about maintaining and deciding how the track should move on. Although I will keep it to myself to l be the person writing the monthly blog posts (which I post on the forum).

Sure, that would be great. I'm happy to keep contributing and to keep learning Crystal :slightly_smiling_face:

**Comment by meatball133** [comment]
> Did we get 5 new exercises? I think I've done 3 (Say, Diamond, Tic-Tac-Toe) and you've done 1 (Bank Account) and I was planning to do the fifth (probably Matrix) this weekend. If you want to push Matrix to next month that's fine with me.

Ohh yeah your right, forgot that, I thought that this was the fifth one. But April isn't over so another exercise this weekend would be great!

> Sure, that would be great. I'm happy to keep contributing and to keep learning Crystal 🙂

Well, I will talk with Erik or Jeremy about adding you as a maintainer, hopefully, shouldn't take too long but will not promise it will happen this week.

**Comment by ryanplusplus** [comment]
Thanks! I should be able to get something up this weekend, hopefully with time to merge before April ends.

**Pull request merged**